plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -376,7 +381,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
-        return getColumns(icebergTable.schema(), typeManager).stream()
+        ImmutableList.Builder<IcebergColumnHandle> columns = ImmutableList.builder();
+        columns.addAll(getColumns(icebergTable.schema(), typeManager));
+        columns.add(IcebergColumnHandle.getPathColumnHandle(FILE_PATH.id()));
```

Comment:
you don't really need to specify here the `id` of the column handle because it is static. We will not have another path column handle with a different id

getPathColumnHandle(id) -> pathColumnHandle()

Reply:
+1

Reply:
@findinpath @alexjo2144 updated